govcd/org_test.go

```diff
@@ -174,6 +175,86 @@ func (vcd *TestVCD) Test_Admin_GetVdcByName(check *C) {
 	check.Assert(err, IsNil)
 }
 
+// Tests org function GetVDCByName with the vdc specified
+// in the config file. Then tests with a vdc that doesn't exist.
+// Fails if the config file name doesn't match with the found vdc, or
+// if the invalid vdc is found by the function.  Also tests an vdc
+// that doesn't exist. Asserts an error if the function finds it or
+// if the error is not nil.
+func (vcd *TestVCD) Test_CreateVdc(check *C) {
+	if vcd.skipAdminTests {
+		check.Skip("Configuration org != 'Sysyem'")
+	}
+
+	results, err := vcd.client.QueryWithNotEncodedParams(nil, map[string]string{
+		"type":   "providerVdc",
+		"filter": fmt.Sprintf("(name==%s)", vcd.config.VCD.ProviderVdc.Name),
+	})
+	check.Assert(err, IsNil)
+	providerVdcHref := results.Results.VMWProviderVdcRecord[0].HREF
+
+	results, err = vcd.client.QueryWithNotEncodedParams(nil, map[string]string{
+		"type":   "providerVdcStorageProfile",
+		"filter": fmt.Sprintf("(name==%s)", vcd.config.VCD.ProviderVdc.StorageProfile),
+	})
+	check.Assert(err, IsNil)
+	providerVdcStorageProfileHref := results.Results.ProviderVdcStorageProfileRecord[0].HREF
+
+	adminOrg, err := GetAdminOrgByName(vcd.client, vcd.org.Org.Name)
+	check.Assert(err, IsNil)
+	check.Assert(adminOrg, Not(Equals), AdminOrg{})
+	vdcConfiguration := &types.VdcConfiguration{
+		Name:            TestCreateOrgVdc,
+		Xmlns:           "http://www.vmware.com/vcloud/v1.5",
+		AllocationModel: "ReservationPool",
+		ComputeCapacity: []*types.ComputeCapacity{
+			&types.ComputeCapacity{
+				CPU: &types.CapacityWithUsage{
+					Units:     "MHz",
+					Allocated: 1024,
+					Limit:     1024,
+				},
+				Memory: &types.CapacityWithUsage{
+					Allocated: 1024,
+					Limit:     1024,
+				},
+			},
+		},
+		VdcStorageProfile: &types.VdcStorageProfile{
+			Enabled: true,
+			Units:   "MB",
+			Limit:   1024,
+			Default: true,
+			ProviderVdcStorageProfile: &types.Reference{
+				HREF: providerVdcStorageProfileHref,
+			},
+		},
+		ProviderVdcReference: &types.Reference{
+			HREF: providerVdcHref,
+		},
+	}
+	task, err := adminOrg.CreateVdc(vdcConfiguration)
```

Comment:
The VDC creation test should do the following:
* test all three allocation models, either re-creating the same VDC or creating three different ones;
* Use both thin provisioning and fast provisioning (we need the tests to use little resources)
* Check that the resulting VDC is enabled and has the relevant elements according to the previous points.

Reply:
Yes. Let's test what is allowed.